iothub/device/src/Transport/AmqpIoT/AmqpIoTTransport.cs

```diff
@@ -0,0 +1,195 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Threading.Tasks;
+using System.Net.Security;
+using System.Security.Cryptography.X509Certificates;
+using System.Net.WebSockets;
+using System.Net;
+using System.Threading;
+using Microsoft.Azure.Devices.Shared;
+using Microsoft.Azure.Amqp;
+using Microsoft.Azure.Amqp.Transport;
+
+namespace Microsoft.Azure.Devices.Client.Transport.AmqpIoT
+{
+    internal class AmqpIoTTransport
+    {
+        private bool _disableServerCertificateValidation;
+        private string _hostName;
+        private AmqpSettings _amqpSettings;
+        private AmqpTransportSettings _amqpTransportSettings;
+        private TlsTransportSettings _tlsTransportSettings;
+
+        public AmqpIoTTransport(AmqpSettings amqpSettings, AmqpTransportSettings amqpTransportSettings, string hostName, bool disableServerCertificateValidation)
+        {
+            _amqpSettings = amqpSettings;
+            _amqpTransportSettings = amqpTransportSettings;
+            _hostName = hostName;
+            _disableServerCertificateValidation = disableServerCertificateValidation;
+
+            var tcpTransportSettings = new TcpTransportSettings()
+            {
+                Host = hostName,
+                Port = AmqpConstants.DefaultSecurePort
+            };
+
+            _tlsTransportSettings = new TlsTransportSettings(tcpTransportSettings)
+            {
+                TargetHost = hostName,
+                Certificate = null,
+                CertificateValidationCallback = _amqpTransportSettings.RemoteCertificateValidationCallback ?? OnRemoteCertificateValidation
+            };
+
+            if (_amqpTransportSettings.ClientCertificate != null)
+            {
+                _tlsTransportSettings.Certificate = _amqpTransportSettings.ClientCertificate;
+            }
+        }
+
+        //STATIC
+        internal async Task<TransportBase> Initialize(TimeSpan timeout)
+        {
+            return await InitializeTransport(timeout).ConfigureAwait(false);
```

Comment:
I don't understand the need for this function.
Also, whenever this is done, it should simply return the Task, not await it and the method shouldn't be marked with `async`.